### DISCUSSION TITLE
Connecting to appium server on a virtual machine

### DISCUSSION THREAD

**Discussion by mostafatouny** [appium/appium#17767] [Q&A]
Hi, I wish to connect appium inspector on my local machine to appium server on a virtual machine. A python script on the virtual machine successfully connected to 10.0.2.15:4723. I have forwarded 127.0.0.1:4723 to 10.0.2.15:4723 on virtual box, but the connection of my local machine to 127.0.0.1:4723 fails.

On my local machine, running appium inspector yields `Could not connect to server; are you sure it's running? If you are using the browser version, also ensure your Appium server has been started with --allow-cors.`

On my local machine, running python `driver = webdriver.Remote("http://127.0.0.1:4723/wd/hub", desired_cap)` yields: `urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))`

Any help?

**Comment by mostafatouny** [comment]
I don't have android sdk on my local machine. Is it required by the inspector?

**Reply by KazuCocoa** [reply]
no, it isn't. I wonder if your inspector send a request to `http://127.0.0.1:4723/`, not with `/wd/hub`. https://github.com/appium/appium-inspector#important-migration-notes

**Reply by mostafatouny** [reply]
The error occurred while I am connecting my inspector with `/wd/hub`

**Reply by KazuCocoa** [reply]
What response can you get when you just call `http://127.0.0.1:4723/wd/hub` without appium? e.g. `curl http://127.0.0.1:4723/wd/hub`

**Reply by mostafatouny** [reply]
@KazuCocoa, On the VM, calling `curl http://10.0.2.15:4723/wd/hub` yields `{"status":9,"value":{"error":"unknown command","message":"The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource","stacktrace":""}}`. On local machine, calling `curl http://127.0.0.1:4723/wd/hub` yields `curl: (56) Recv failure: Connection reset by peer`

Again, running python script `driver = webdriver.Remote("http://10.0.2.15:4723/wd/hub", desired_cap)` on the VM launches google calculator on my phone!

**Reply by jlipps** [reply]
try `curl http://10.0.2.15:4723/wd/hub/status` then

**Reply by mostafatouny** [reply]
@jlipps On the VM, calling `curl http://10.0.2.15:4723/wd/hub/status` yields `{"value":{"build":{"version":"1.22.3"}},"sessionId":null,"status":0}` but on my local machine calling `curl http://127.0.0.1:4723/wd/hub/status` yields `curl: (56) Recv failure: Connection reset by peer`

**Reply by KazuCocoa** [reply]
> I have forwarded 127.0.0.1:4723 to 10.0.2.15:4723 on virtual box

Not sure how you started appium, but what about starting an appium server with `-a 0.0.0.0` host?

e.g. https://superuser.com/questions/448941/accessing-ubuntu-guest-virtualbox-node-js-server-from-windows-host

**Reply by mostafatouny** [reply]
@KazuCocoa, On my VM, Calling `curl http://0.0.0.0:4723/wd/hub/status` yields `{"value":{"build":{"version":"1.22.3"}},"sessionId":null,"status":0}`. On my local machine, Calling `curl http://127.0.0.1:4723/wd/hub/status` yields `curl: (56) Recv failure: Connection reset by peer`. On my virtual box, `127.0.0.1:4723` is forwarded to `0.0.0.0 4723`

**Comment by jlipps** [comment]
i meant to try `curl http://10.0.2.15:4723/wd/hub/status` from your local machine. can it access the vm at `10.0.2.15`?

**Reply by mostafatouny** [reply]
My local machine can't access `10.0.2.15`. I must manually add an entry for port-forwarding on virtualbox. Maybe there's another port or address that is invoked implicitly which I need to forward too?
